From f1fd531496bd6b7e4329d5d11bc95a1ae8300391 Mon Sep 17 00:00:00 2001
From: "Sreeves, Francis" <francis.sreeves@digi.com>
Date: Thu, 21 May 2020 16:49:19 +0100
Subject: [PATCH] user/gpsd: poll for one packet at a time

Exit the device polling loop when a full packet has been received and
parsed, allowing other devices to be serviced - and gpsd clients to
connect - even if the device delivers a full packet at every single
read, as does, for example, the Navilock NL-82012U external USB device.

DAL-2577
---
 gpsd/libgpsd_core.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gpsd/libgpsd_core.c b/gpsd/libgpsd_core.c
index 47ee5d57e03a..1b652f05f959 100644
--- a/gpsd/libgpsd_core.c
+++ b/gpsd/libgpsd_core.c
@@ -2078,7 +2078,6 @@ int gpsd_multipoll(const bool data_ready,
                 handler(device, changed);
             }
 
-#ifdef __future__
             // this breaks: test/daemon/passthrough.log ??
             /*
              * Bernd Ocklin suggests:
@@ -2094,7 +2093,6 @@ int gpsd_multipoll(const bool data_ready,
             if (0 != (changed & PACKET_SET)) {
                break;
             }
-#endif  // __future__
         }
     } else if (0 < device->reawake &&
                time(NULL) > device->reawake) {
-- 
2.34.1

